app/src/test/java/io/apicurio/registry/noprofile/serde/JsonSchemaSerdeTest.java

```diff
@@ -134,6 +134,54 @@ public void testJsonSchemaSerde() throws Exception {
         }
     }
 
+    @Test
+    public void testJsonSchemaSerdeAutoRegister() throws Exception {
+        String groupId = TestUtils.generateGroupId();
+        String artifactId = generateArtifactId();
+
+        Person person = new Person("Carles", "Arnal", 30);
+
+        try (JsonSchemaKafkaSerializer<Person> serializer = new JsonSchemaKafkaSerializer<>(restClient, true);
+             Deserializer<Person> deserializer = new JsonSchemaKafkaDeserializer<>(restClient, true)) {
+
+            Map<String, Object> config = new HashMap<>();
+            config.put(SerdeConfig.EXPLICIT_ARTIFACT_GROUP_ID, groupId);
+            config.put(SerdeConfig.ARTIFACT_RESOLVER_STRATEGY, SimpleTopicIdStrategy.class.getName());
+            config.put(SerdeConfig.SCHEMA_LOCATION, "/io/apicurio/registry/util/json-schema.json");
+            config.put(SerdeConfig.AUTO_REGISTER_ARTIFACT, true);
+            serializer.configure(config, false);
+
+            deserializer.configure(Collections.emptyMap(), false);
+
+            Headers headers = new RecordHeaders();
+            byte[] bytes = serializer.serialize(artifactId, headers, person);
+
+            person = deserializer.deserialize(artifactId, headers, bytes);
+
+            Assertions.assertEquals("Carles", person.getFirstName());
+            Assertions.assertEquals("Arnal", person.getLastName());
+            Assertions.assertEquals(30, person.getAge());
+
+            person.setAge(-1);
+
+            try {
+                serializer.serialize(artifactId, new RecordHeaders(), person);
+                Assertions.fail();
```

Comment:
There is `Assertions.assertThrows`

Reply:
It's not throwing an exception, if it gets to this point, I want to force the test to fail, as it's happening with `Assertions.fail()`

Reply:
argh,you're right